datacenterlight/tasks.py

```diff
@@ -94,33 +80,42 @@ def create_vm_task(self, vm_template_id, user, specs, template,
         if vm_id is None:
             raise Exception("Could not create VM")
 
-        # Create a Hosting Order
-        order = HostingOrder.create(
-            price=final_price,
-            vm_id=vm_id,
-            customer=customer,
-            billing_address=billing_address
+        # Update HostingOrder with the created vm_id
+        hosting_order = HostingOrder.objects.filter(id=order_id).first()
+        error_msg = None
+        if hosting_order:
```

Comment:
Just a style question - but wouldn't it be more pythonic to just try to update and try ... except the error?

Reply:
agree !